api/tsup.config.js

```diff
@@ -0,0 +1,11 @@
+import cpy from 'cpy';
+import { defineConfig } from 'tsup';
+
+export default defineConfig({
+	entry: ['src/index.ts', 'src/cli/run.ts'],
+	format: 'esm',
+	async onSuccess() {
+		await cpy(['src/**/*.{yaml,liquid}'], 'dist');
+	},
+	dts: 'src/index.ts',
+});
```

Comment:
this is effectively the workaround you mentioned right? Shall we add a comment as a reminder why this is needed with a reference to the issue  https://github.com/evanw/esbuild/pull/2508 ? 🤔

Reply:
No, this one is just a 1-to-1 replacement of the previous build script:
```json
	"build": "tsc --project tsconfig.prod.json && copyfiles \"src/**/*.{yaml,liquid}\" -u 1 dist",
```

The actual workarounds are applied at multiple places:

- **Workaround 1** - Context-depended paths for getting `yaml` / `liquid` files:
  - [`api/src/services/mail/index.ts`](https://github.com/directus/directus/pull/19089/files#diff-b65b152ce847c4566f72298678c2903bcb0f123ca8de71abb5743bd3e0be7b4d)
  - [`api/src/cli/utils/create-db-connection.ts`](https://github.com/directus/directus/pull/19089/files#diff-b65bd1e3d47d761595d4de9024f0df5773d9907ff99f68f76c20def71e47459b)
  - [`api/src/cli/utils/create-env/index.ts`](https://github.com/directus/directus/pull/19089/files#diff-8e6b4acbcb5a8b312a01b1c3d4bd156a5cff84dbc1ec862fac7554eb4f6bf8f8)
  - [`api/src/database/system-data/fields/index.ts`](https://github.com/directus/directus/pull/19089/files#diff-026ca6355fa8884f6e25bfc7f6cc8351ec67dfc33a5c674b8e8bf09793b00945)
  - [`api/src/database/system-data/collections/index.ts`](https://github.com/directus/directus/pull/19089/files#diff-ab42e6dd9097a8bb51184962cb4d5d08df30da719c7a41f0f2499d77baf935ca)
  - [`api/src/database/system-data/app-access-permissions/index.ts`](https://github.com/directus/directus/pull/19089/files#diff-5798bdac1e9a9acc4f624c0dbd8dbeb3eb18cae164b673108602577f7ae3b60d)
  - [`api/src/database/system-data/relations/index.ts`](https://github.com/directus/directus/pull/19089/files#diff-512f6b0837b355105727fa449a6abd84ff9b37139bccaa52b2d152c6ce758a45)
  - [`api/src/database/seeds/run.ts`](https://github.com/directus/directus/pull/19089/files#diff-ea5259e27174c13eb7e51ecdb72b21d57236f1fef9ddb95efb5ec7572707c833)
  All files are using `CONTEXT_ROOT` from [api/src/constants.ts](https://github.com/directus/directus/pull/19089/files#diff-5da24c390f8befd890a3c9e96c17862e386efabee8937b09120547bfcaa397d0) which is where the path is computed once using [findContextRoot](https://github.com/directus/directus/pull/19089/files#diff-e47968a1a045eda0ed8db3525151bb8a5cb7962067c87d8dba2f355af7e24fb4). So maybe we could add comment in this file?

- **Workaround 2** - Statically imported scripts:
  - Direct import of internal migrations ([`9875f1e`](https://github.com/directus/directus/pull/19089/commits/9875f1ebb62a9dac4c2d8c564785adeda69a0f89))
  -  Direct import of internal operations ([`b3bc237`](https://github.com/directus/directus/pull/19089/commits/b3bc237b1690fc72c9ac35282db3d085c6226861))
  Would you like to have a comment in both places?

Reply:
I see 👍 👍 well, or better just a linear issue to keep track of it?

Reply:
Good idea! Added to backlog 👍